macro/src/main/scala/monocle/macros/Focus.scala

```diff
@@ -0,0 +1,35 @@
+package monocle.macros
+
+import scala.language.dynamics
+
+import monocle._
+import monocle.syntax.ApplyLens
+
+/** Syntactic sugar around an ApplyLens */
+class Focus[S,A](val applyLens: ApplyLens[S,S,A,A]) extends AnyVal with Dynamic{
+  def apply[C](field: A => C): Focus[S,C] = macro internal.FocusImpl.apply[S,A,C]
+
+  /**
+  Whitebox macro for field access syntax.
+  E.g. someobj.focus.a.b...
+  */
+  def selectDynamic(field: String): Any = macro internal.FocusImpl.selectDynamicImpl
+  
+  /**
+  Assignment syntax for updates
+  E.g. someobj.focus(_.a.b) = ...
+  */
+  def update[C](field: A => C, value: C): S = macro internal.FocusImpl.update[S,A,C]
+  /**
+  Assignment syntax for updates with member access syntax
+  E.g. someobj.focus.a.b = ...
+  */
+  def updateDynamic[C](field: String)(value: C): S = macro internal.FocusImpl.updateDynamicImpl[S,C]
```

Comment:
oh I didn't know this feature. I tried on a basic case class but it does not compile:
```scala
case class Foo(i: Int){
    def update(newI: Int): Foo = copy(i = newI)
  }

  val foo = Foo(3)
  Foo(6) = 5
```